### PR TITLE
[stable9] fix searchbox background

### PR DESCRIPTION
**Pull request by jancborchardt**
@MorrisJobke can you fix the unit tests? I don’t know which exact features are in there to be tested at that point, like which have been backported. Also cc @juliushaertl for info since some of the inversion stuff etc was done by you. :)

Please review @nextcloud/designers @mar1u5 @Bugsbane @williambargent @juliushaertl @MorrisJobke

### CODE REVIEW

**Comment by jancborchardt** [comment]
@MorrisJobke assigned you to check for the tests. :)

**Comment by MorrisJobke** [comment]
> @MorrisJobke assigned you to check for the tests. :)

I'm not your minion: You could at least check the output and fix the easy stuff https://travis-ci.org/nextcloud/server/jobs/147977236#L297 first. Then if you don't know how to fix the rest I could help you.

File: apps/theming/tests/lib/controller/ThemingControllerTest.php [hunk 1]
```diff
@@ -314,6 +314,36 @@ public function testGetStylesheetWithOnlyColor() {
 			->expects($this->at(1))
 			->method('getAppValue')
 			->with('theming', 'color', '')
+<<<<<<< HEAD
+=======
```
Comment (MorrisJobke) [inline apps/theming/tests/lib/controller/ThemingControllerTest.php]: *looks-angry-to-Jan*

Comment (jancborchardt) [inline apps/theming/tests/lib/controller/ThemingControllerTest.php]: Well so what’s the deal – are the new additions the relevant ones, or should the change just be the empty one? You’re not my minion, but I’m also not a PHP coder. ;)

Comment (jancborchardt) [inline apps/theming/tests/lib/controller/ThemingControllerTest.php]: That is, of course I looked at the conflicts and tried fixing them before opening this pull request. But I have no idea what parts of the PHP code are new and which old. Just poking in the dark by removing stuff here and other stuff there doesn’t seem like a good idea, that’s why I asked for your help.

Comment (MorrisJobke) [inline apps/theming/tests/lib/controller/ThemingControllerTest.php]: Should work now.

File: apps/theming/tests/lib/controller/ThemingControllerTest.php [hunk 2]
```diff
@@ -314,6 +314,33 @@ public function testGetStylesheetWithOnlyColor() {
...
+			->method('getAppValue')
+			->with('theming', 'backgroundMime', '')
+			->willReturn('');
+
+		$expected = new Http\DataDownloadResponse('#body-user #header,#body-settings #header,#body-public #header,#body-login,.searchbox input[type="search"]:focus,.searchbox input[type="search"]:active,.searchbox input[type="search"]:valid {background-color: #000}', 'style', 'text/css');
+		$expected->cacheFor(3600);
+		@$this->assertEquals($expected, $this->themingController->getStylesheet());
+	}
+
+	public function testGetStylesheetWithOnlyColorInvert() {
```
Comment (juliushaertl) [inline apps/theming/tests/lib/controller/ThemingControllerTest.php]: This is a duplicate unit test, as the color inverting was not backported to Nc9, but I guess this doesn't hurt.

File: apps/theming/tests/lib/controller/ThemingControllerTest.php [hunk 3]
```diff
@@ -416,7 +443,43 @@ public function testGetStylesheetWithAllCombined() {
...
+				background-image: url(\'./logo?v=0\');
+				background-size: contain;
+			}#body-login {
+				background-image: url(\'./loginbackground?v=0\');
+			}', 'style', 'text/css');
+		$expected->cacheFor(3600);
+		@$this->assertEquals($expected, $this->themingController->getStylesheet());
+	}
+
+	public function testGetStylesheetWithAllCombinedInverted() {
```
Comment (juliushaertl) [inline apps/theming/tests/lib/controller/ThemingControllerTest.php]: same as above.

**Comment by juliushaertl** [comment]
I guess we can keep the unit tests, even if they are kind of duplicate.

:+1: from my side

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Comment by jancborchardt** [comment]
Thanks for your help @MorrisJobke! :) Maybe a talk on getting into writing tests at the Nextcloud Conference would be cool ;)

**Pull request merged**